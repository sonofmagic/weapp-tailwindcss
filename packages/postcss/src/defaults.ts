import type { IStyleHandlerOptions } from './types'

export function getDefaultOptions(_options?: Partial<IStyleHandlerOptions>): Partial<IStyleHandlerOptions> {
  return {
    // https://github.com/postcss/postcss-calc
    cssPresetEnv: {
      features: {
        'cascade-layers': true,
        'is-pseudo-class': {
          specificityMatchingName: 'weapp-tw-ig',
        },
        'oklab-function': true,
        'color-mix': true,
        // 在 calc 下，这个需要开启
        'custom-properties': false,
        //  options?.cssPresetEnv?.features?.['custom-properties'] ?? options?.cssCalc
        //   ? { preserve: false }
        //   : false,
      },
      autoprefixer: {
        add: false,
      },
    },
    // 支付宝小程序不支持，所以默认关闭
    cssRemoveProperty: true,
    // cssRemoveAtSupports: true,
    // cssRemoveAtMedia: true,
    cssSelectorReplacement: {
      root: 'page',
      universal: ['view', 'text'],
    },
  }
}
