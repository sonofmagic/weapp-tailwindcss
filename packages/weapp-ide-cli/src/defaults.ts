import path from 'node:path'
import os from 'node:os'

const homedir = os.homedir()
const SupportedPlatformsMap = {
  Windows_NT: 'Windows_NT',
  Darwin: 'Darwin',
}

const defaultPathMap = {
  [SupportedPlatformsMap.Windows_NT]:
    'C:\\Program Files (x86)\\Tencent\\微信web开发者工具\\cli.bat',
  [SupportedPlatformsMap.Darwin]:
    '/Applications/wechatwebdevtools.app/Contents/MacOS/cli',
} as Record<string, string>
export const operatingSystemName = os.type()

export const defaultCustomConfigDirPath = path.join(homedir, '.weapp-ide-cli')
export const defaultCustomConfigFilePath = path.join(
  defaultCustomConfigDirPath,
  'config.json',
)

export const defaultPath = defaultPathMap[operatingSystemName]
