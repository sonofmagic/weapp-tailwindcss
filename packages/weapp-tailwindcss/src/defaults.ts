import type { AppType, UserDefinedOptions } from './types'
import { isAllowedClassName, MappingChars2String } from '@weapp-core/escape'
import { isPackageExists } from 'local-pkg'
import { noop } from './utils'

const CSS_FILE_PATTERN = /.+\.(?:wx|ac|jx|tt|q|c|ty)ss$/
const HTML_FILE_PATTERN = /.+\.(?:(?:wx|ax|jx|ks|tt|q|ty|xhs)ml|swan)$/
const JS_FILE_PATTERN = /.+\.[cm]?js?$/

const MAIN_CSS_CHUNK_MATCHERS: Partial<Record<AppType, (file: string) => boolean>> = {
  'uni-app': file => file.startsWith('common/main') || file.startsWith('app'),
  'uni-app-vite': file => file.startsWith('app') || file.startsWith('common/main'),
  'mpx': file => file.startsWith('app'),
  'taro': file => file.startsWith('app'),
  'remax': file => file.startsWith('app'),
  'rax': file => file.startsWith('bundle'),
  'native': file => file.startsWith('app'),
  'kbone': file => /^(?:common\/)?miniprogram-app/.test(file),
}

const alwaysFalse = () => false

function createMainCssChunkMatcher() {
  return (file: string, appType?: AppType) => {
    if (!appType) {
      return true
    }
    const matcher = MAIN_CSS_CHUNK_MATCHERS[appType]
    return matcher ? matcher(file) : true
  }
}

export function getDefaultOptions(): UserDefinedOptions {
  return {
    /**
     * wxss 微信小程序
     * acss 支付宝小程序
     * jxss 京东小程序
     * ttss 头条小程序
     * qss QQ小程序
     * css 最正常的样式文件
     * tyss 涂鸦小程序
     */
    cssMatcher: file => CSS_FILE_PATTERN.test(file),
    /**
     * wxml 微信小程序
     * axml 支付宝小程序
     * jxml 京东小程序
     * ksml 快手小程序
     * ttml 头条小程序
     * qml QQ小程序
     * tyml 涂鸦小程序
     * xhsml 小红书小程序
     * swan 百度小程序
     */
    htmlMatcher: file => HTML_FILE_PATTERN.test(file),
    jsMatcher: (file) => {
      if (file.includes('node_modules')) {
        return false
      }
      // remove jsx tsx ts case
      return JS_FILE_PATTERN.test(file)
    },
    mainCssChunkMatcher: createMainCssChunkMatcher(),
    wxsMatcher: alwaysFalse,
    // https://tailwindcss.com/docs/preflight#border-styles-are-reset-globally
    cssPreflight: {
      'box-sizing': 'border-box',
      'border-width': '0',
      'border-style': 'solid',
      'border-color': 'currentColor',
    },

    disabled: false,
    customRuleCallback: noop,
    onLoad: noop,
    onStart: noop,
    onEnd: noop,
    onUpdate: noop,

    customAttributes: {},
    customReplaceDictionary: MappingChars2String,
    appType: undefined,
    arbitraryValues: {
      allowDoubleQuotes: false,
    },
    cssChildCombinatorReplaceValue: ['view', 'text'],
    inlineWxs: false,
    injectAdditionalCssVarScope: false,
    jsPreserveClass: (keyword) => {
      /**
       * 默认保留 keyword
       */
      if (keyword === '*') {
        return true
      }
      return false
    },
    disabledDefaultTemplateHandler: false,
    cssSelectorReplacement: {
      root: ['page', '.tw-root'],
      universal: ['view', 'text'],
    },
    babelParserOptions: {
      sourceType: 'unambiguous',
      cache: true,
    },
    postcssOptions: {},
    // 开发版本微信小程序工具和小米13手机上 @property 是有效果的
    // 支付宝小程序会直接挂掉
    // https://developer.mozilla.org/en-US/docs/Web/CSS/@property
    cssRemoveProperty: true,
    cssRemoveHoverPseudoClass: true,
    ignoreCallExpressionIdentifiers: isPackageExists('@weapp-tailwindcss/merge') ? ['twMerge', 'twJoin', 'cva', 'tv'] : [],
    ignoreTaggedTemplateExpressionIdentifiers: ['weappTwIgnore'],
    replaceRuntimePackages: false,
    tailwindcssPatcherOptions: {
      filter(className: string) {
        return !isAllowedClassName(className)
      },
    },
    logLevel: 'info',
  }
}
