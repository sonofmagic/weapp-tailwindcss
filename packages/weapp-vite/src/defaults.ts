import type { WatchOptions } from './types'

export const defaultExcluded: string[] = ['**/node_modules/**', '**/miniprogram_npm/**']

export function getWeappWatchOptions(): WatchOptions {
  return {
    paths: ['**/*.{wxml,json,wxs}', '**/*.{png,jpg,jpeg,gif,svg,webp}', '.env', '.env.*'],
    ignored: [
      ...defaultExcluded,
    ],
  }
}
