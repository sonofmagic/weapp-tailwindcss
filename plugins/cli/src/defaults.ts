import type { Result } from 'postcss-load-config'
import type { BuildOptions } from './type'

const defaultJavascriptExtensions = ['js'] // , 'cjs', 'mjs']

const defaultTypescriptExtensions = ['ts'] // , 'cts', 'mts']

const defaultWxsExtensions = ['wxs']

export const defaultNodeModulesDirs = [
  '**/node_modules/**',
  '**/miniprogram_npm/**',
  '**/project.config.json/**',
  '**/project.private.config.json/**',
  '**/package.json/**',
  'postcss.config.js',
  'tailwind.config.js',
  'weapp-tw.config.js',
]

function noop(): undefined {}

export function getDefaultOptions(options?: Partial<BuildOptions>, postcssOptionsFromConfig?: Result | undefined): Partial<BuildOptions> {
  return {
    outDir: 'dist',
    weappTailwindcssOptions: {},
    clean: true,
    src: '',
    exclude: [...defaultNodeModulesDirs],
    include: ['**/*.{png,jpg,jpeg,gif,svg,webp}'],
    extensions: {
      javascript: [...defaultJavascriptExtensions, ...defaultTypescriptExtensions, ...defaultWxsExtensions],
      html: ['wxml'],
      css: ['wxss', 'less', 'sass', 'scss'],
      json: ['json'],
    },
    watchOptions: {
      cwd: options?.root,
      events: ['add', 'change', 'unlink', 'ready'],
    },
    postcssOptions: postcssOptionsFromConfig,
    gulpChain: noop,
  }
}
