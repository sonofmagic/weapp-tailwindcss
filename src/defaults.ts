import type { UserDefinedOptions } from './types'
import { noop } from '@/utils'
import { SimpleMappingChars2String } from '@/escape'

// import { mangleClassRegex } from '@/mangle/expose'

export const defaultOptions: UserDefinedOptions = {
  cssMatcher: (file) => /.+\.(?:wx|ac|jx|tt|q|c)ss$/.test(file),
  htmlMatcher: (file) => /.+\.(?:(?:(?:wx|ax|jx|ks|tt|q)ml)|swan)$/.test(file),
  jsMatcher: (file) => {
    if (file.includes('node_modules')) {
      return false
    }
    // remove jsx tsx ts case
    return /.+\.[cm]?js?$/.test(file)
  },
  mainCssChunkMatcher: (file, appType) => {
    switch (appType) {
      case 'uni-app': {
        return /^common\/main/.test(file)
      }
      case 'uni-app-vite': {
        // vite 旧版本和新版本对应的样式文件
        return file.startsWith('app') || /^common\/main/.test(file)
      }
      case 'mpx': {
        return file.startsWith('app')
      }
      case 'taro': {
        return file.startsWith('app')
      }
      case 'remax': {
        return file.startsWith('app')
      }
      case 'rax': {
        return file.startsWith('bundle')
      }
      case 'native': {
        return file.startsWith('app')
      }
      case 'kbone': {
        return /^(?:common\/)?miniprogram-app/.test(file)
      }
      default: {
        return true
      }
    }
  },
  wxsMatcher: () => {
    return false
  },
  // https://tailwindcss.com/docs/preflight#border-styles-are-reset-globally
  cssPreflight: {
    'box-sizing': 'border-box',
    'border-width': '0',
    'border-style': 'solid',
    'border-color': 'currentColor'
  },

  disabled: false,
  customRuleCallback: noop,
  onLoad: noop,
  onStart: noop,
  onEnd: noop,
  onUpdate: noop,

  customAttributes: {},
  customReplaceDictionary: SimpleMappingChars2String,

  supportCustomLengthUnitsPatch: {
    units: ['rpx'],
    dangerousOptions: {
      gteVersion: '3.0.0',
      lengthUnitsFilePath: 'lib/util/dataTypes.js',
      packageName: 'tailwindcss',
      variableName: 'lengthUnits',
      overwrite: true
    }
  },
  appType: undefined,
  arbitraryValues: {
    allowDoubleQuotes: false
  },
  cssChildCombinatorReplaceValue: ['view'],
  inlineWxs: false,
  injectAdditionalCssVarScope: false,
  jsPreserveClass: (keyword) => {
    /**
     * 默认保留 keyword
     */
    if (keyword === '*') {
      return true
    }
    return false
  },
  disabledDefaultTemplateHandler: false,
  cssSelectorReplacement: {
    root: 'page',
    universal: ['view', 'text']
  },
  babelParserOptions: {
    sourceType: 'unambiguous'
  },
  postcssOptions: {}
  // jsAstTool: 'ast-grep'
}
