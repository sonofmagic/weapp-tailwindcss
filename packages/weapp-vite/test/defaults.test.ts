import { defaultExcluded, getWeappWatchOptions } from '@/defaults'

describe('defaults', () => {
  it('getWeappWatchOptions', () => {
    expect(getWeappWatchOptions()).toMatchSnapshot()
  })

  it('defaultExcluded', () => {
    expect(defaultExcluded).toMatchSnapshot()
  })

  // it('getDefaultViteConfig case 0', () => {
  //   const ctx = createContext()
  //   const config = getDefaultViteConfig(ctx)
  // })
})
