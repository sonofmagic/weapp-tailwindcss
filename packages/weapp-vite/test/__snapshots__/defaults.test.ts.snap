// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`defaults > defaultExcluded 1`] = `
[
  "**/node_modules/**",
  "**/miniprogram_npm/**",
]
`;

exports[`defaults > getWeappWatchOptions 1`] = `
{
  "ignored": [
    "**/node_modules/**",
    "**/miniprogram_npm/**",
  ],
  "paths": [
    "**/*.{wxml,json,wxs}",
    "**/*.{png,jpg,jpeg,gif,svg,webp}",
    ".env",
    ".env.*",
  ],
}
`;
